### ISSUE TITLE
Add Hydrological Data to dashboards as downloadable csv files

### ISSUE THREAD

**Issue by helendallas** [kartoza/django-bims#2838]
Once the Hydrological Regions and Flow Regime Types have been added to FBIS, please can a new section be added to each Dashboard where **Hydrological Data** can be downloaded as a csv file.

Below the current Abiotic data section in all dashboards.

All 144 values in the Flow type metrics.xlsx file should be shown for filtered sites in the selected Quin_IDs

**Comment by dimasciput** [comment]
![image](https://user-images.githubusercontent.com/1979569/150310350-30bd8a30-a8be-47ff-82c4-2b46db13aa31.png)
 Add a download button here

**Comment by dimasciput** [comment]
![image](https://user-images.githubusercontent.com/1979569/150310704-631f0002-3a45-490e-a584-a1393b73948e.png)

**Comment by dimasciput** [comment]
this is in testing

**Comment by helendallas** [comment]
Thanks @dimasciput 

I had hoped only the data for the quinary clicked by the user would download and that @amyburness could provide this file of the attribute data associated with each Hydrological Region Flow Type. 

But if this is difficult to do, then just leave it as is. Hopefully the user will figure out to look up the quinary id, and codes in the consolidated file that is downloaded.

**Comment by amyburness** [comment]
@helendallas  and @dimasciput 

For this data there are 2 version on the database one that has all of the data included and one that is the summary based on the summary data sent in the form of a materialised view of the larger dataset. 

The materialised view (MV) it what is being visualised when the user click on the map so that there are not 100s of data to read through

@helendallas do you want the user to download the full data for each point oth the MV?

**Comment by helendallas** [comment]
Yes @amyburness When the user clicks on a quinary they get the attributes shown on the screen, but it would be useful for the user to be able to then  download the full data for each point. If this is a mission then is it possible for them to be able to download the MV for a point that they have clicked?

**Comment by amyburness** [comment]
@dimasciput 
If I put the full table on geoserver can you make the request pull from that one instead of the MV?

**Comment by amyburness** [comment]
@dimasciput the full dataset is on GeoServer here this is what the download should point to so that the user gets the full data set,  [kartoza:flow_regime_type_full](https://maps.kartoza.com/geoserver/web/wicket/bookmarkable/org.geoserver.web.data.resource.ResourceConfigurationPage?name=flow_regime_type_full&wsName=kartoza)

**Comment by helendallas** [comment]
@dimasciput Any progress?

**Comment by helendallas** [comment]
@dimasciput @amyburness This is still just a huge download of all data, rather than a download of data for the quinary selected.

**Comment by helendallas** [comment]
@amyburness PLease cna you see what the solution is here. thanks

**Comment by helendallas** [comment]
@dimasciput as discussed in a sprint planning meeting, this will take half a day. I am going to put this ticket on hold for now as there are more urgent tickets needing to be sorted first.

**Comment by helendallas** [comment]
@dimasciput Please update this based on work done on pesticides, thanks

**Comment by dimasciput** [comment]
@helendallas this has been added to testing, please test

**Comment by helendallas** [comment]
@dimasciput It is not very useful to users when they download the csv, without the correct headers showing what the values are. Can you please make the column headers the same as what you see when you click on the map, rather than A2, A3.....M1, M2...etc

<img width="963" alt="Screenshot 2023-10-24 at 08 44 54" src="https://github.com/kartoza/django-bims/assets/45286876/b586bc38-c4bb-4388-8f54-b5bd7a103a77">

**Comment by dimasciput** [comment]
@helendallas, here is the master file I use. You can simply update the header. Would you like me or Amy to handle the update?
[Flow type metrics.csv](https://github.com/kartoza/django-bims/files/13111598/Flow.type.metrics.csv)

@amyburness, do you perhaps have the updated version?

**Comment by helendallas** [comment]
Ah ok, If @amyburness doesn't have one I can update and send. @amyburness please let me know

**Comment by helendallas** [comment]
@dimasciput @amyburness 

I have a better idea. Is it possible to include the pdf with the download csv file, in a separate tab, or whatever way makes sense so that a user can know what the column headers are.

Here is the pdf.

[Flow Group metric definitions.pdf](https://github.com/kartoza/django-bims/files/13113296/Flow.Group.metric.definitions.pdf)

**Comment by dimasciput** [comment]
@helendallas yes I can add that.

**Comment by amyburness** [comment]
@dimasciput I sent you the CSV let me know if you need anything further from me.

**Comment by dimasciput** [comment]
@helendallas I have added a button to download the document, you can also update it yourself here https://testing.healthyrivers.kartoza.com/admin/bims/nonbiodiversitylayer/49/change/

**Comment by helendallas** [comment]
That's brilliant @dimasciput Very useful, thanks. Good to deploy to production

**Comment by dimasciput** [comment]
@helendallas deployed to production

**Comment by helendallas** [comment]
Thanks @dimasciput Closing ticket
